src/viewport.js

```diff
@@ -124,32 +124,44 @@ export default class Viewport {
    * @param {Array} lngLatZ - [lng, lat] or [lng, lat, Z]
    * @param {Object} opts - options
    * @param {Object} opts.topLeft=true - Whether projected coords are top left
-   * @return {Array} - [x, y] or [x, y, z] in top left coords
+   * @return {Array} - screen coordinates [x, y] or [x, y, z], z as pixel depth
    */
-  project(xyz, {topLeft = false} = {}) {
+  project(xyz, {topLeft = true} = {}) {
     const [x0, y0, z0 = 0] = xyz;
     assert(Number.isFinite(x0) && Number.isFinite(y0) && Number.isFinite(z0), ERR_ARGUMENT);
 
     const [X, Y] = this.projectFlat([x0, y0]);
     const v = transformVector(this.pixelProjectionMatrix, [X, Y, z0, 1]);
 
     const [x, y] = v;
-    const y2 = topLeft ? this.height - y : y;
-    return xyz.length === 2 ? [x, y2] : [x, y2, 0];
+    const y2 = topLeft ? y : this.height - y;
```

Comment:
looks like we need change the doc instead of change the code here, cause current code make that while topLeft == true, the output is actually bottomLeft based.

Reply:
`pixelProjectionMatrix` has y already flipped. We are not going to change the matrix because we are matching it 1:1 with mapbox-gl.